importer/src/main/resources/db/migration/v1/V1.110.0__fix_transfer_synthetic_contract_logs.sql

```diff
@@ -0,0 +1,48 @@
+with entity_addresses as (
+    select num, evm_address
+    from entity
+    where length(evm_address) > 0
+),
+cl_ext as (
+    select *,
+           encode(topic1, 'hex') as topic1_hex,
+           encode(topic2, 'hex') as topic2_hex
+    from contract_log
+    where topic0 = '\xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef'
+      and (
+          octet_length(topic1) < 20 or
+          octet_length(topic2) < 20 or
+          substr(topic1, 1, 12) = repeat(e'\\000', 12)::bytea or
+          substr(topic2, 1, 12) = repeat(e'\\000', 12)::bytea
+      )
+)
+
+update contract_log cl
+set topic1 = coalesce(e1.evm_address, cl.topic1),
+    topic2 = coalesce(e2.evm_address, cl.topic2)
+from cl_ext
+left join entity_addresses e1 on e1.num = (
```

Comment:
there's no index on `num`, and so far in mainnet, we always have `entity.id == entity.num`

```suggestion
left join entity_addresses e1 on e1.id = (
```

Reply:
Can we make this change and assume it will never be ran on shard/realm != 0?

Reply:
yeah. If a third party runs a really old release on a non-zero realm network, they will find out later they have to restart with a compatible version.

If it's a release the fix and ran on a non-zero realm network, the tables should be empty and the migration shouldn't change any data.